python3/libexec/nbd_client_manager.py

```diff
@@ -1,52 +1,66 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 """
 Provides functions and a CLI for safely connecting to and disconnecting from
 NBD devices.
 """
 
 import argparse
+import fcntl
+import json
 import logging
 import logging.handlers
 import os
+import re
 import subprocess
 import time
-import fcntl
-import json
-import re
 from datetime import datetime, timedelta
 
-
 LOGGER = logging.getLogger("nbd_client_manager")
 LOGGER.setLevel(logging.DEBUG)
 
-LOCK_FILE = '/var/run/nonpersistent/nbd_client_manager'
+LOCK_FILE = "/var/run/nonpersistent/nbd_client_manager"
 
 # Don't wait more than 10 minutes for the NBD device
 MAX_DEVICE_WAIT_MINUTES = 10
 
 
+class NotGetNbdNumber(Exception):
+    """
+    The NBD device should be in this format: nbd{0-100}
+    If we cannot match this pattern, raise this exception
+    """
+
+class NbdConnStateTimeout(Exception):
+    """
+    If we cannot get the connection status of a nbd device,
+    raise this exception.
+    """
+
 class NbdDeviceNotFound(Exception):
     """
     The NBD device file does not exist. Raised when there are no free NBD
     devices.
     """
+
     def __init__(self, nbd_device):
-        super(NbdDeviceNotFound, self).__init__(
-            "NBD device '{}' does not exist".format(nbd_device))
+        super().__init__(
+            "NBD device '{}' does not exist".format(nbd_device)
+        )
         self.nbd_device = nbd_device
 
-
-class FileLock(object):
+class FileLock: # pragma: no cover
     """Container for data relating to a file lock"""
+
     def __init__(self, path):
         self._path = path
         self._lock_file = None
 
     def _lock(self):
         """Acquire the lock"""
         flags = fcntl.LOCK_EX
-        self._lock_file = open(self._path, 'w+')
+        # pylint: disable=consider-using-with
```

Comment:
can not use `with` state here as it is a file lock, need to close the file at proper place explicitly.
So it make sense to me to disable this check.

Reply:
The `with` will be applied to `FileLock`.